### PR TITLE
ContextProvider

### PR DESCRIPTION
**Pull request by wtfil**
Doc https://github.com/wixplosives/stylable-components-guide/blob/master/code/api/context-provider.md

### CODE REVIEW

File: src/components/context-provider/context-provider.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    style?: object;
+    children?: React.ReactNode;
+    [key: string]: any;
+}
+
+const fakeChildContextTypes = new Proxy({}, {
+    has() {return true; }
+});
+
+function omit<T extends {[key: string]: any}>(obj: T, ...skip: string[]): Partial<T> {
```
Comment (tyv) [inline src/components/context-provider/context-provider.tsx]: this should be in utils

**Comment by wtfil** [comment]
@idoros 
I did not like how `ContextProvider` manipulates with prop names.
```tsx
'context-foo' => 'foo'
'foo' => 'context-provider-foo'
Symbol => Symbol // this actually will not work
```

I would like to see all off these properties as is and also to have them under the single container:

```tsx
'foo' => 'contextProvider.foo'
```

i.e.

```ts
class ContextProvider extends React.Component {
    static childContextTypes = {
        contextProvider: PropTypes.object
    }
    getChildContext() {
        return {
            contextProvider: {......}
        }
    }
}
```

this API will be much cleaner and also allows to get rid of `Proxy` which I used to pass the [React's](https://github.com/facebook/react/blob/65b9ad94aaf62eb61b7b3888362c2e7c0b06697a/src/renderers/shared/fiber/ReactFiberContext.js#L199) context check

File: src/components/context-provider/context-provider.tsx (hunk 1 above)
Comment (wtfil) [reply src/components/context-provider/context-provider.tsx]: moved

**Comment by tyv** [comment]
@wtfil let's go on with your proposal. 
@idoros agreed.

**Comment by wtfil** [comment]
@tyv api simplified, you can review once again

File: src/components/context-provider/context-provider.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import * as PropTypes from 'prop-types';
+import * as React from 'react';
+import {omit} from '../../utils';
+
+export interface CPProps {
```
Comment (tyv) [inline src/components/context-provider/context-provider.tsx]: maybe name it `ContextProviderProps` — i don't see any reason to make acronym here.

Comment (wtfil) [reply src/components/context-provider/context-provider.tsx]: `+`

**Comment by tyv** [comment]
Lets add docs!

**Comment by wtfil** [comment]
added doc and test for nested `ContextProvider`

File: test/components/context-provider.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+                </ContextProvider>
+            ));
+            inner1 = renderer.select('TEST_DIV_1');
+            inner2 = renderer.select('TEST_DIV_2');
+        });
+
+        it('first Inner component should have dir="ltr', () => {
+            expect(inner1).attr('dir', 'ltr');
+        });
+        it('first Inner component should have contnt "10"', () => {
```
Comment (tyv) [inline test/components/context-provider.spec.tsx]: contnt > content

File: test/components/context-provider.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            expect(inner1).text('10');
+        });
+        it('second Inner component should have dir="rtl', () => {
+            expect(inner2).attr('dir', 'rtl');
+        });
+        it('second Inner component should have contnt "20"', () => {
```
Comment (tyv) [inline test/components/context-provider.spec.tsx]: contnt > content

**Pull request merged**